### PR TITLE
Add Banca d'Italia exchange rates extension

### PR DESCRIPTION
**Pull request by FezVrasta**
## Description

Allows to convert currencies using the "Banca d'Italia" exchange rates. This can be useful in case an Italian company needs to emit an invoice that will be paid in a currency different than euro, since it's often easier to emit the invoice in euro, converted from the non-euro currency, and if this happens, the "Banca d'Italia" exchange rates are the ones that must be used.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/5382443/150979245-9e6b6f3a-8d02-406e-b8f4-307cdd252af6.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/banca-d-italia-currency-converter/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+type Rate = {
+  country: string;
+  currency: string;
+  ISOCode: string;
+  rate: number;
+};
+
+const ENDPOINT =
+  "https://tassidicambio.bancaditalia.it/terzevalute-wf-web/rest/v1.0/dailyRates?currencyIsoCode=EUR&lang=en";
```
Comment (FezVrasta) [inline extensions/banca-d-italia-currency-converter/src/index.tsx:24]: I'm using the same endpoint used by the official web-based conversion tool (https://tassidicambio.bancaditalia.it/terzevalute-wf-ui-web/converter). Unfortunately they use CSV as format... 🤷‍♂️

File: extensions/banca-d-italia-currency-converter/package.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  "name": "banca-d-italia-currency-converter",
+  "title": "Banca d'Italia Currency Converter",
+  "description": "Convert currencies based on the Banca d'Italia exchange rates.",
+  "icon": "command-icon.png",
+  "author": "FezVrasta",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Convert Currency",
```
Comment (pernielsentikaer) [inline extensions/banca-d-italia-currency-converter/package.json:12]:
```suggestion
      "title": "Convert Currency",
      "subtitle": "Banca d'Italia",
```

File: extensions/banca-d-italia-currency-converter/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      }
+    }
+  }, [fromAmount, toCurrency, fromCurrency, rates]);
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <CopyToClipboardAction content={toAmount} />
+          <ActionPanel.Item
```
Comment (pernielsentikaer) [inline extensions/banca-d-italia-currency-converter/src/index.tsx:88]: You're missing an icon for this action

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @FezVrasta 👋 

Thanks for your contribution, I left some feedback for you to address 💪 

You could consider showing something else than “NaN” when you delete the value from Amount?

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @FezVrasta 👋 

Now when opening the extension there is an error, also the dropdown is only showing EUR, can you look into this?

vy8rKT5XW0WvT061

le6Doyt9c5H0CvWn

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by FezVrasta** [comment]
Notice the popup on the bottom? There aren't conversion rates for the specified date

**Comment by pernielsentikaer** [comment]
lHi @FezVrasta 🔥 

Touché! I just thought that it was an error since the rates should be able for today, or is that normal? Expect me to be the end user using it 😄 

I'm not sure if it makes any sense, but you could hide the form field when the selected date is not able to find rates and use a `Form.Description` (added in ^1.27.1)

**Comment by FezVrasta** [comment]
I want to avoid the jarring experience of having the field appear and disappear as the user changes dates

**Review by pernielsentikaer — APPROVED** [review]
Hi @FezVrasta 👋 

Thanks for your contribution, let's get to the last stage 🚢  🔥

**Pull request merged**